protostar/starknet_gateway/gateway_facade_test.py

```diff
@@ -0,0 +1,214 @@
+            ],
+            "name": "InnerStruct",
+            "size": 3,
+            "type": "struct"
+        },
+        {
+            "members": [
+                {
+                    "name": "low",
+                    "offset": 0,
+                    "type": "felt"
+                },
+                {
+                    "name": "high",
+                    "offset": 1,
+                    "type": "felt"
+                }
+            ],
+            "name": "Uint256",
+            "size": 2,
+            "type": "struct"
+        },
+        {
+            "inputs": [
+                {
+                    "name": "amount",
+                    "type": "felt"
+                }
+            ],
+            "name": "increase_balance",
+            "outputs": [],
+            "type": "function"
+        },
+        {
+            "inputs": [],
+            "name": "get_balance",
+            "outputs": [
+                {
+                    "name": "res",
+                    "type": "felt"
+                }
+            ],
+            "stateMutability": "view",
+            "type": "function"
+        },
+        {
+            "inputs": [
+                {
+                    "name": "arg",
+                    "type": "Test"
+                },
+                {
+                    "name": "felt_arg",
+                    "type": "felt"
+                },
+                {
+                    "name": "uint256_arg",
+                    "type": "Uint256"
+                },
+                {
+                    "name": "felt_list_arg_len",
+                    "type": "felt"
+                },
+                {
+                    "name": "felt_list_arg",
+                    "type": "felt*"
+                },
+                {
+                    "name": "inst_list_arg_len",
+                    "type": "felt"
+                },
+                {
+                    "name": "inst_list_arg",
+                    "type": "InnerStruct*"
+                }
+            ],
+            "name": "constructor",
+            "outputs": [],
+            "type": "constructor"
+        }
+    ]
+}
+"""
+
+
+def test_validating_deploy_inputs_pass():
+    inputs = [
+        0x0,  # Test.felt_field
+        0xA,  # Test.inner_field.a
+        0xB,  # Test.inner_field.b
+        0xC,  # Test.inner_field.c
+        0x9,  # felt_arg
+        0x1,  # Uint256.low
+        0x2,  # Uint256.high
+        5,  # len(felt_list_arg)
+        0xA,  # |
+        0xB,  # |
+        0xC,  # |
+        0xD,  # |
+        0xE,  # | felt* (felt_list_arg)
+        2,  # len(inst_list_arg)
+        0xA,  # inst_list_arg[0].a
+        0xB,  # inst_list_arg[0].b
+        0xC,  # inst_list_arg[0].c
+        0x1,  # inst_list_arg[1].a
+        0x2,  # inst_list_arg[1].b
+        0x3,  # inst_list_arg[1].c
+    ]
+
+    validate_deploy_input(COMPILED_CONTRACT_ABI_ONLY, inputs)
+
+
+def test_validating_deploy_inputs_not_enough():
```

Comment:
I would also add a test for empty inputs list.

Reply:
done